core/src/main/java/io/aiven/kafka/tieredstorage/RemoteStorageManager.java

```diff
@@ -301,35 +355,30 @@ private void uploadSegmentLog(final RemoteLogSegmentMetadata remoteLogSegmentMet
         }
     }
 
-    private void uploadIndexFile(final RemoteLogSegmentMetadata remoteLogSegmentMetadata,
-                                 final InputStream index,
-                                 final IndexType indexType,
-                                 final SegmentEncryptionMetadata encryptionMetadata,
-                                 final SegmentCustomMetadataBuilder customMetadataBuilder)
-        throws StorageBackendException, IOException {
-        TransformChunkEnumeration transformEnum = new BaseTransformChunkEnumeration(index);
+    private InputStream transformIndex(final IndexType indexType,
+                                       final InputStream index,
+                                       final int size,
+                                       final SegmentEncryptionMetadata encryptionMetadata,
+                                       final SegmentIndexesV1Builder segmentIndexBuilder) {
+        TransformChunkEnumeration transformEnum = new BaseTransformChunkEnumeration(index, Integer.MAX_VALUE / 2);
```

Comment:
Will this work?
```suggestion
        TransformChunkEnumeration transformEnum = new BaseTransformChunkEnumeration(index, size);
```

Reply:
Actually, it does! Thanks for the suggestion.